### ISSUE TITLE
Function barrier_touched compares log returns (ret) to normal returns (target)?

### ISSUE THREAD

**Issue by quoctuams** [hudson-and-thames/mlfinlab#201]
**Describe the bug**
The output of `ml.labeling.get_bins` sometimes show bin = 0 eventhough ret > trgt.
Looking at the function `labeling.barrier_touched`, I noticed that the param `out_df['ret']` is the log return calculated in get_bins, and that param is compared with target to compute the bins:
```
pt_level_reached = ret > target * events.loc[date_time, 'pt']
sl_level_reached = ret < -target * events.loc[date_time, 'sl']
```
The target is usually a multipler of daily volatility, as shown in the book, and it is calculated using normal returns. 
So is the comparison above intended? Should the target be calculated using log return instead?

The `out_df['ret']` later is transformed back to normal return and shown along the `trgt` as result of `get_bins` function. If the `trgt` is intended to be log then there is no need to transform the `ret` to be normal returns. 

Thanks,

**Comment by Jackal08** [comment]
Hi @quoctuams, 

Thanks for highlighting this. The reason we take the log of the returns is to create position symmetry between long and short positions. If your short and price drops 2% you would have made an amount of profit slightly different to that. Taking the log of returns solves that problem. 

The bug your raising here should be corrected and should be a relatively to easy fix. Would you mind to please make a PR for this?

The reason we left this in the past is that the marginal difference between the target and log target should be very small, this does, of course, result in slightly different labels. It should be fixed.

**Comment by quoctuams** [comment]
> The reason we left this in the past is that the marginal difference between the target and log target should be very small, this does, of course, result in slightly different labels. It should be fixed.

Thanks for clarifying. In my case the investment horizon is in weeks rather than days so the difference in target and log target turned out to be quite large, leading to some events mislabeled. 

> The bug your raising here should be corrected and should be a relatively to easy fix. Would you mind to please make a PR for this?
The fix should be to log transform the target to compare with log returns, then output the normal target along with normal returns. I'm not familiar with how PR works but will try to see what I can do this weekend. Thanks :)

**Comment by Jackal08** [comment]
> The fix should be to log transform the target to compare with log returns, then output the normal target along with normal returns. I'm not familiar with how PR works but will try to see what I can do this weekend. Thanks :)

Great, thank you very much! Ping us if you need help.

**Comment by Jackal08** [comment]
I think the following line of code would correct this? 

Original
```
pt_level_reached = ret > target * events.loc[date_time, 'pt']
sl_level_reached = ret < -target * events.loc[date_time, 'sl']
```

New
```
pt_level_reached = ret > np.log(1 + target) * events.loc[date_time, 'pt']
sl_level_reached = ret < -np.log(1 + target) * events.loc[date_time, 'sl']
```

**Comment by quoctuams** [comment]
Yes it is 👍

**Comment by MislavSag** [comment]
I have this problem too. Here us my output table from get_bins function:


ret             | trgt          | bin | side
0.006406  | 0.005344 | 1    | -1.0
0.005716  | 0.005504 | 1    | -1.0
0.013032  | 0.005164 | 1    | -1.0
-0.019153 | 0.009673 | 0    | -1.0
-0.021063 | 0.009941 | 0    | -1.0

You can see in the 4th row that "ret" is lower than "trgt", which means bin should be -1, am I right? I think this also changes the result of the notebook in research repo:  https://github.com/hudson-and-thames/research/blob/master/Chapter3/2019-03-06_JJ_Trend-Follow-Question.ipynb

**Comment by quoctuams** [comment]
Bin only takes values [0,1], with 0 means you should not take that bet. So
in this case your results look fine to me.

The original bug only appears when the target is big enough that the normal
returns and log returns differ significantly. Sorry I havent got time to
make a PR to fix it. Perhaps someone else familiar with how PR works can
help? The fix is quite easy tbh :(

**Comment by MislavSag** [comment]
I thought it should be 1 and -1, but this is only if I omit side argument. I get it now :)

I changed above code in fork, but wait for my other pull request to be accepted (if it would be). I will than add change you and @Jackal08 proposed.

**Comment by quoctuams** [comment]
Great thank you!

**Comment by Jackal08** [comment]
Fixed in #223
